src/status.rs

```diff
@@ -309,9 +427,370 @@ mod test {
+        )
+    }
+
+    #[test]
+    fn leader_has_no_messages() {
+        let approve_state = get_approve_state_msg();
+        let heartbeat = get_heartbeat_msgs(Duration::minutes(0));
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Leader has no new messages but the follower has heartbeats and approvestate"
+        )
+    }
+
+    #[test]
+    fn follower_has_no_messages() {
+        let heartbeat = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Follower has no new messages but leader has heartbeats and newstate"
+        )
+    }
+
+    #[test]
+    fn both_arrays_have_messages() {
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(0));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat_1]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat_2]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            false,
+            "Both arrays have messages"
+        )
+    }
+
+    // is_disconnected()
+    #[test]
+    fn no_recent_hbs_on_both_sides() {
+        let server = SERVER_POOL.get_server();
```

Comment:
Since `is_disconnected` doesn't make a server request like `is_finalized`, you can `clone` the `DUMMY_CHANNEL` directly instead of starting a mock server and using `get_test_channel`.